### ISSUE TITLE
Problems with Travis + Prisma ^1.7.x

### ISSUE THREAD

**Issue by KATT** [prismagraphql/prisma#2437]
# Bug Report / (potentially just help needed)

## tl;dr

1. Use Travis
2. `docker compose up -d`
3. `prisma deploy`

Travis is unable to find local cluster. (_You're not logged in and cluster local could not be found locally. Trying to authenticate._ on `prisma deploy`)

Started occurring when updating from prisma `1.6.3` to `1.7.1` (before that a docker config file wasn't needed, so seems like some changes have been made there)

## Reproduction

- Link to travis output: https://travis-ci.org/KATT/shop/builds/378963929?utm_source=github_status&utm_medium=notification
- Link to PR with failing prisma upgrade + failing build: https://github.com/KATT/shop/pull/93

<details><summary>Travis output</summary>

```
[..]

$ docker-compose -f api/database/docker-compose.yml up -d
Creating network "database_default" with the default driver
Creating volume "database_postgres" with default driver
Pulling prisma (prismagraphql/prisma:1.8)...
1.8: Pulling from prismagraphql/prisma
Digest: sha256:922a533d68d23c02ab248971e6a076c2840d2a98efff21fcfcc3882254661ec8
Status: Downloaded newer image for prismagraphql/prisma:1.8
Pulling postgres (postgres:latest)...
latest: Pulling from library/postgres
Digest: sha256:23c92d48934b254c007c08812638acfd849fe1c179a34b63c752a62100c6524d
Status: Downloaded newer image for postgres:latest
Creating database_prisma_1 ... 
Creating database_postgres_1 ... 
Creating database_postgres_1
Creating database_prisma_1
$ cd api && yarn prisma deploy
yarn run v1.3.2
$ /home/travis/build/[secure]/shop/api/node_modules/.bin/prisma deploy
You're not logged in and cluster local could not be found locally. Trying to authenticate.
Authenticating...
Opening https://app.prisma.io/cli-auth?secret=$2a$08$TWT3BgS415eJCPuWQsGZmO in the browser

```
</details>

---

Is there something I'm missing in the docker-compose or in the `prisma deploy`-command to find the local cluster?

As it worked with Prisma on `1.6.x` I don't _think_ it's a problem with Travis.

**Comment by SpaceK33z** [comment]
I'm pretty sure you're running into #2310, which has been fixed in the latest version.

**Comment by KATT** [comment]
Thanks for the answer @SpaceK33z. I suspect that as well, but [I'm using `1.8.3`](https://github.com/KATT/shop/blob/renovate/prisma-1.x/api/package.json#L56) which is the latest published on npm. Or do I need to update the [`image`](https://github.com/KATT/shop/blob/renovate/prisma-1.x/api/database/docker-compose.yml#L4) in the docker config or something?

**Comment by marktani** [comment]
You need to update both the CLI and the Docker image 🙂

**Comment by KATT** [comment]
Will do later / tmrw!

**Comment by KATT** [comment]
I set the docker image [to explicit `1.8.3`](https://github.com/KATT/shop/pull/93/commits/1b47577e50279a47e4087a86d8122f5b4516ddcf#diff-5019039d1c883494dd59b696ab489069L4) but it didn't seem to make any difference

<details>

```
217.26s$ cd api && yarn prisma deploy
yarn run v1.3.2
$ /home/travis/build/[secure]/shop/api/node_modules/.bin/prisma deploy
You're not logged in and cluster local could not be found locally. Trying to authenticate.
Authenticating...
Opening https://app.prisma.io/cli-auth?secret=$2a$08$Z3iQ99b4/NhdZ89nEHVZFO in the browser
Authenticating... !
ERROR: GraphQL Error (Code: 502)
```
</details>

**Comment by divyenduz** [comment]
@KATT - I was able to get this to working by using the following configuration:-

[Config file used](https://github.com/divyenduz/ds/blob/18e57f3b2270b5c7952f00ce2cb1ffabecb19d20/.travis.yml)

[Passing build with Prisma deploy](https://travis-ci.org/divyenduz/ds/builds/394943504) 

It would be awesome if you convert [this example](https://github.com/divyenduz/ds) to a new [Prisma example](https://github.com/prismagraphql/prisma/tree/master/examples) in a PR

**Comment by divyenduz** [comment]
@KATT : I think you also need to update your `prisma.yml` to deploy to localhost. The login step is what block you from running it on travis and login should not be asked for when deploying locally.

**Comment by KATT** [comment]
Added `endpoint:` in the `prisma.yml`

https://github.com/KATT/shop/pull/93/commits/558d61af46fe97f0d54b7e6ff964c42635e8c798

**Comment by KATT** [comment]
🎉 

Whoop whoop @divyenduz -- adding `endpoint:` seemed to have done the trick. It goes past that stage now. It breaks later but whatever is happening it's not because of this issue anymore! I'll do an update when I have a neat setup. 🎉 

Thank you for your help @divyenduz and @marktani !

**Comment by marktani** [comment]
I put together a minimal example for a Travis setup here: https://github.com/prismagraphql/prisma/tree/master/examples/travis 🙂

**Comment by roycclu** [comment]
@divyenduz hey man, from your config files and job log, it looks like you are spinning up fresh set of docker processes. And then seeding them with data. 

What about in a CD environment? Wouldn't you just keep one mysql instance and one prisma instance and continuously deploy new schema to existing instance? 

Has your team come across this?

**Comment by divyenduz** [comment]
@roycclu : Based on your environment/use-case you can modify your `docker-compose.yml` to make prisma talk to an ever running `mysql` backend via environment variables or by hard-coding. Does that make sense? Happy to help.

**Comment by roycclu** [comment]
@divyenduz thanks man.

It seems docker-compose is more useful for local environments. Using volume should persist data.

And in a CI/CD environment, would you say it's better to set up the initial prisma service manually and then run a "prisma deploy" script, in CI migration stage, after the node server is built and deployed?

Basically I'm deciding between setting up and running a new prisma server every git push. Or continuously deploy to an initial manually set up prisma server. How are you guys managing this?
